doc/arch/adr-004-geoprocessing-api.md

```diff
@@ -0,0 +1,56 @@
+## 004 - Expose Geoprocessing API
+
+### Context
+The MMW/BigCZ platform performs a number of geospatial processing tasks which can be initiated by submitting tasks to an HTTP endpoint.  The web application provides the tools and workflow for assembling and submitting the input, monitoring execution progress, and collecting and visualizing the results of these jobs.  The platform provides value by assembling the required national scale data and the infrastructure necessary to perform the processing. In an effort to extend this capability to alternate workflows that don't involve the web platform, the client wants to formalize an externally consumable HTTP API.
+
+MMW/BigCZ already includes and makes limited use of the Django REST Framework (DRF), which supports convenience methods around creating HTTP APIs.
+
+#### API Objectives
+* Support execution of all current and future "Analyze" sub tasks (Land Use, Soil, etc)
+* Support execution of Rapid Watershed Delineation
+* Model execution functionality will not be exposed
+* Available only for registered users
+* Usage will be recorded
+* Long term goal would be to charge for access
+
+#### Considerations
```

Comment:
I think that it would be valuable to consider adding Swagger support from the start. Not just for documentation (which we'll probably need to supply in some capacity), but also for aiding in API design.